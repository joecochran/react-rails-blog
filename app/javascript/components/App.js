import React from 'react';
import { Route, Switch, Link } from 'react-router-dom';

import Home from './Home';
import Posts from './Posts';
import NewPost from './NewPost';
import Post from './Post';
import '../stylesheets/App.scss';

class App extends React.Component {
  render() {
    return (
      <div>
        <nav className="nav">
          <div className="container">
            <ul className="nav__list">
              <li className="nav__item"><Link to="/" className="nav__link">Home</Link></li>
              <li className="nav__item"><Link to="/posts" className="nav__link">Posts</Link></li>
              <li className="nav__item"><Link to="/new_post" className="nav__link">New Post</Link></li>
            </ul>
          </div>
        </nav>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/posts" component={Posts} />
          <Route exact path="/posts/:id" component={Post} />
          <Route exact path="/new_post" component={NewPost} />
        </Switch>
      </div>
    )
  }
}

export default App;
