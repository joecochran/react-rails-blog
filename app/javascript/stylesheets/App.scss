html {
  box-sizing: border-box;
}

*, *:before, *:after {
  box-sizing: inherit;
}

body {
  margin: 0;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: 300;
}

.container {
  max-width: 900px;
  margin: 0 auto;
}

.nav {
  padding: 1em;
  background: linear-gradient(120deg, #32cec3 1%,#36b4c7 50%,#3f96d8 100%);

  &__list {
    display: flex;
    list-style: none;
    margin: 0;
    padding: 0;
  }

  &__item {
    margin-right: 1em;
  }

  &__link {
    color: #fff;
  }
}

.post {
  &__header {

  }
}
