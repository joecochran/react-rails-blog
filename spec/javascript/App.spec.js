import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from 'components/App';

configure({ adapter: new Adapter() });

describe('home component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
