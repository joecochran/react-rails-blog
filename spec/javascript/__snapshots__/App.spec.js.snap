// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          className="bg-teal p-6"
        >
          <div
            className="container mx-auto"
          >
            <ul
              className="list-reset flex"
            >
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin"
                >
                  Dashboard
                </NavLink>
              </li>
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin/posts"
                >
                  Posts
                </NavLink>
              </li>
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin/new_post"
                >
                  New Post
                </NavLink>
              </li>
            </ul>
          </div>
        </nav>,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/admin"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/admin/posts"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/admin/posts/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/admin/new_post"
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container mx-auto"
          >
            <ul
              className="list-reset flex"
            >
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin"
                >
                  Dashboard
                </NavLink>
              </li>
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin/posts"
                >
                  Posts
                </NavLink>
              </li>
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin/new_post"
                >
                  New Post
                </NavLink>
              </li>
            </ul>
          </div>,
          "className": "bg-teal p-6",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="list-reset flex"
            >
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin"
                >
                  Dashboard
                </NavLink>
              </li>
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin/posts"
                >
                  Posts
                </NavLink>
              </li>
              <li
                className="mr-6"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="text-teal-lightest"
                  to="/admin/new_post"
                >
                  New Post
                </NavLink>
              </li>
            </ul>,
            "className": "container mx-auto",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin"
                  >
                    Dashboard
                  </NavLink>
                </li>,
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/posts"
                  >
                    Posts
                  </NavLink>
                </li>,
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/new_post"
                  >
                    New Post
                  </NavLink>
                </li>,
              ],
              "className": "list-reset flex",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin"
                  >
                    Dashboard
                  </NavLink>,
                  "className": "mr-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Dashboard",
                    "className": "text-teal-lightest",
                    "to": "/admin",
                  },
                  "ref": null,
                  "rendered": "Dashboard",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/posts"
                  >
                    Posts
                  </NavLink>,
                  "className": "mr-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Posts",
                    "className": "text-teal-lightest",
                    "to": "/admin/posts",
                  },
                  "ref": null,
                  "rendered": "Posts",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/new_post"
                  >
                    New Post
                  </NavLink>,
                  "className": "mr-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "New Post",
                    "className": "text-teal-lightest",
                    "to": "/admin/new_post",
                  },
                  "ref": null,
                  "rendered": "New Post",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/admin"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/admin/posts"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/admin/posts/:id"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/admin/new_post"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/admin",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/admin/posts",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/admin/posts/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/admin/new_post",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            className="bg-teal p-6"
          >
            <div
              className="container mx-auto"
            >
              <ul
                className="list-reset flex"
              >
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin"
                  >
                    Dashboard
                  </NavLink>
                </li>
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/posts"
                  >
                    Posts
                  </NavLink>
                </li>
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/new_post"
                  >
                    New Post
                  </NavLink>
                </li>
              </ul>
            </div>
          </nav>,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/admin"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/admin/posts"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/admin/posts/:id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/admin/new_post"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container mx-auto"
            >
              <ul
                className="list-reset flex"
              >
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin"
                  >
                    Dashboard
                  </NavLink>
                </li>
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/posts"
                  >
                    Posts
                  </NavLink>
                </li>
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/new_post"
                  >
                    New Post
                  </NavLink>
                </li>
              </ul>
            </div>,
            "className": "bg-teal p-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="list-reset flex"
              >
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin"
                  >
                    Dashboard
                  </NavLink>
                </li>
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/posts"
                  >
                    Posts
                  </NavLink>
                </li>
                <li
                  className="mr-6"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="text-teal-lightest"
                    to="/admin/new_post"
                  >
                    New Post
                  </NavLink>
                </li>
              </ul>,
              "className": "container mx-auto",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="mr-6"
                  >
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="text-teal-lightest"
                      to="/admin"
                    >
                      Dashboard
                    </NavLink>
                  </li>,
                  <li
                    className="mr-6"
                  >
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="text-teal-lightest"
                      to="/admin/posts"
                    >
                      Posts
                    </NavLink>
                  </li>,
                  <li
                    className="mr-6"
                  >
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="text-teal-lightest"
                      to="/admin/new_post"
                    >
                      New Post
                    </NavLink>
                  </li>,
                ],
                "className": "list-reset flex",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="text-teal-lightest"
                      to="/admin"
                    >
                      Dashboard
                    </NavLink>,
                    "className": "mr-6",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "aria-current": "page",
                      "children": "Dashboard",
                      "className": "text-teal-lightest",
                      "to": "/admin",
                    },
                    "ref": null,
                    "rendered": "Dashboard",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="text-teal-lightest"
                      to="/admin/posts"
                    >
                      Posts
                    </NavLink>,
                    "className": "mr-6",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "aria-current": "page",
                      "children": "Posts",
                      "className": "text-teal-lightest",
                      "to": "/admin/posts",
                    },
                    "ref": null,
                    "rendered": "Posts",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="text-teal-lightest"
                      to="/admin/new_post"
                    >
                      New Post
                    </NavLink>,
                    "className": "mr-6",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "aria-current": "page",
                      "children": "New Post",
                      "className": "text-teal-lightest",
                      "to": "/admin/new_post",
                    },
                    "ref": null,
                    "rendered": "New Post",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/admin"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/admin/posts"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/admin/posts/:id"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/admin/new_post"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/admin",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/posts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/posts/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/admin/new_post",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
